src/API.Abstraction/Cache/CacheSerializationHelper.cs

```diff
@@ -0,0 +1,49 @@
+﻿using ProtoBuf;
+using System.Reflection;
+
+namespace API.Abstraction.Cache
+{
+	public static class CacheSerializationHelper
+	{
+		public static byte[] GetBytes<T>(T entity)
+		{
+			ValidateType<T>();
+			using var memoryStream = new MemoryStream();
+			Serializer.Serialize(memoryStream, entity);
+			return memoryStream.ToArray();
+		}
+
+		public static bool TryGetFromBytes<T>(byte[] bytes, out T entity)
+		{
+			entity = default;
+			try
+			{
+				ValidateType<T>();
+				using var memoryStream = new MemoryStream(bytes);
+				var result = Serializer.Deserialize<T>(memoryStream);
+				entity = result;
+				return true;
+			}
+			catch (Exception)
+			{
+				return false;
+			}
+		}
```

Comment:
This implementation looks pretty heavy - try-catch + this ValudateType method using reflexy - for each call... But you don't have a simple check for entity = null

Reply:
Putting this null check here would give you the ability to skip this check in all places where you call this method.

Reply:
Yeah, it really simplifies things in `GetOrCreate`.